### DISCUSSION TITLE
Plans to rework internals around the notification system

### DISCUSSION THREAD

**Discussion by wax911** [AniTrend/anitrend-app#420] [Ideas]
I'm creating this discussion to add onto https://github.com/AniTrend/anitrend-app/pull/419#issuecomment-842507963

As it stands the current implementation around notifications allows the user to read notifications on anilist and mark them as read on the site and have the application still show an unread indicator on them. The same could be done for AT to some extent to prevent constant notification reminders if the notification contents have not changed, e.g. instead of storing all the read notification ids we could just store the last read **Id** and last received **Id** but allow the user decide what happens when they swipe the notification away. I believe this would achieve what you want @youssefhabri right?

I'm going to tag @taichikuji @luk1337 to see if they have any other ideas

**Comment by luk1337** [comment]
>instead of storing all the read notification ids we could just store the last read Id and last received Id

Well...currently, we are always sending notifications for `min(unread, 5)` notifications even if one of them in that set is marked as read on anilist so we could just store these 5 ids.

**Reply by luk1337** [reply]
Or better, if we continue with `min(unread, 5)` we could just store top id and nothing else.

**Reply by wax911** [reply]
> Or better, if we continue with `min(unread, 5)` we could just store top id and nothing else.

That's what I was thinking too

**Reply by luk1337** [reply]
I think that'd be fine and relatively low effort.

**Reply by luk1337** [reply]
Then we'd also have to rename dismiss option to "mark as read on dismiss" ^^

**Comment by luk1337** [comment]
@wax911 have you ever noticed that clear notifications work just doesn't seem to run after some time? Maybe there are some background execution restrictions in newer Android versions...
I placed breakpoints everywhere and it appears that the job reaches the scheduling part just fine but never hits doWork() in ClearNotificationWorker.

**Reply by luk1337** [reply]
I've had an idea that maybe replacing `ExistingWorkPolicy.KEEP` with `ExistingWorkPolicy.REPLACE` will help in some way or at least if we do that then it'll just clear notifications immediately in case previously scheduled work failed and is awaiting the backoff period.

**Reply by wax911** [reply]
I bad never tested this enough times to notice, If it works around the issue then we can give it shot. Sorry for the late reply I'm not at home and only going to be back tomorrow
